fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/source/lookup/FlinkAsyncLookupFunction.java

```diff
@@ -171,9 +172,9 @@ private void fetchResult(
                         if (row == null) {
                             resultFuture.complete(Collections.emptyList());
                         } else {
-                            // TODO: we can project fluss row first,
-                            //  to avoid deserialize unnecessary fields
-                            RowData flinkRow = flussRowToFlinkRowConverter.toFlinkRowData(row);
+                            RowData flinkRow =
+                                    flussRowToFlinkRowConverter.toFlinkRowData(
```

Comment:
`flussRowToFlinkRowConverter` works on full fields and it fails `ArrayIndexOutOfBoundsException` if passing projected row. Try run `com.alibaba.fluss.connector.flink.source.FlinkTableSourceITCase#testLookup1PkTable`. 

`flussRowToFlinkRowConverter` should also adapted for projection if we want it work on projected row.

Reply:
Thank you, it has been modified, and FlinkTableSourceITCase#testLookup1PkTable has been successfully run